src/slam_karto.cpp

```diff
@@ -280,8 +301,62 @@ SlamKarto::SlamKarto() :
     mapper_->setParamUseResponseExpansion(use_response_expansion);
 
   // Set solver to be used in loop closure
-  solver_ = new SpaSolver();
-  mapper_->SetScanSolver(solver_);
+  std::stringstream solver_plugin_stream, default_solver_plugin_stream;
+  solver_plugin_stream << "karto_plugins::" << solver_type_;
+  default_solver_plugin_stream << "karto_plugins::" << default_solver_type_;
+  try
+  {
+    solver_ = solver_loader_.createInstance(solver_plugin_stream.str());
+    ROS_INFO_STREAM("Loaded " << solver_plugin_stream.str() << " solver plugin");
+  }
+  catch (pluginlib::PluginlibException& ex)
+  {
+    ROS_ERROR_STREAM(
+        "The solver plugin " << solver_plugin_stream.str() << " failed to load for some reason. Error: " << ex.what()
+            << ". We will try to load the default solver plugin instead: " << default_solver_plugin_stream.str());
+    try
+    {
+      solver_ = solver_loader_.createInstance(default_solver_plugin_stream.str());
+    }
+    catch (pluginlib::PluginlibException& ex)
+    {
+      ROS_ERROR_STREAM(
```

Comment:
@stwirth Note that there isn't really a fallback behavior if you fail to load the default solver or visualizer.